doc/porting_processing.dox

```diff
@@ -0,0 +1,72 @@
+- Derive your algorithms from the new base class QgsProcessingAlgorithm (or a subclass of QgsProcessingAlgorithm), not GeoAlgorithm
+
+- Ensure that your algorithm (or algorithm's parent class) implements the new pure virtual createInstance(self, config) call to return a new instance of the algorithm class. Note that this should not return a copy of the algorithm, but instead a newly constructed instance of the class. If you use a base class in your plugin for all algorithms, you can usually shortcut the createInstance implementation by placing:
+ 
+    def createInstance(self, config={}):
+        return type(self)()
+
+  inside your algorithm base class.
+
+- Input parameters and available outputs must be declared in an implementation of the new pure virtual method initAlgorithm(self, config={})
+
+- The input parameters and outputs classes have been replaced with new c++ versions, which must be used when calling addParameter and addOuput.    
+Map parameters from old classes to new classes
+   - ParameterNumber -> QgsProcessingParameterNumber
+       - constructor arguments in different order
+	   - type specified explicitly, not via min/max/default type
+   - ParameterBoolean -> QgsProcessingParameterBoolean
+   - ParameterCrs -> QgsProcessingParameterCrs
+
+- Use featuresources and sinks wherever possible
+
+Update outputs
```

Comment:
As since 058271b0c1 there is no need to declare some outputs explicitly, some instructions how to reference these automatically added outputs (for example to retrieve output file name) are needed.

Also I think we need more detailed instructions about outputs, as in some algs they are added as parameters (like `self.addParameter(QgsProcessingParameterRasterDestination(self.OUTPUT, 'My output'))
`) and other added as outputs (like `self.addOutput(...)`)
```